src/main/java/org/broadinstitute/hellbender/tools/walkers/groundtruth/GroundTruthScorer.java

```diff
@@ -0,0 +1,796 @@
+
+        // cycle skip
+        final FlowBasedReadUtils.CycleSkipStatus cycleSkipStatus = FlowBasedReadUtils.getCycleSkipStatus(flowRead, referenceContext);
+
+        // accumulate reports
+        if ( cycleSkipStatus != FlowBasedReadUtils.CycleSkipStatus.CS && qualReport != null ) {
+            addToQualReport(flowRead, referenceContext, errorProb);
+        }
+
+        // emit
+        try {
+            emit(flowRead, flowHaplotype, score, normalizedScore, errorProb, read, cycleSkipStatus);
+        } catch (IOException e) {
+            throw new GATKException("failed to write output record", e);
+        }
+    }
+
+    private boolean filter(final GATKRead read, final ReferenceContext referenceContext) {
+
+        // loop on features contained in the read, check that they are in agreement with read data
+        Iterator<VariantContext>    iter = features.query(new SimpleInterval(read));
+        byte[]                      ref = referenceContext.getBases();
+        while ( iter.hasNext() ) {
+            final VariantContext vc = iter.next();
+            for ( int refCoord = vc.getStart() ; refCoord <= vc.getEnd() ; refCoord++ ) {
+
+                // get byte from read
+                Optional<Byte>      readByte = ReadUtils.getReadBaseAtReferenceCoordinate(read, refCoord);
+                if ( !readByte.isPresent() ) {
+                    return false;
+                }
+
+                // get byte from reference
+                byte                refByte = ref[refCoord - referenceContext.getWindow().getStart()];
+
+                // compare
+                if ( refByte != readByte.get() ) {
+                    return false;
+                }
+            }
+        }
+
+        // if here, no interference detected
+        return true;
+    }
+
+    /*
+     * compute error probability vector for a read
+     *
+     * The vector has one element for each flow key, representing the probability complementing the call-probability to 1
+     * This is further complicated by the optional presence of a genome-prior database, which provides factoring for
+     * each hmer length (on a base basis)
+     */
+    private double[] computeErrorProb(final FlowBasedRead flowRead, final GenomePriorDB genomePriorDB) {
+
+        final int[] key = flowRead.getKey();
+        final byte[] flowOrder = flowRead.getFlowOrderArray();
+
+        final double[] probCol = new double[flowRead.getMaxHmer() + 1];
+        double[] result = new double[key.length];
+
+        for ( int i = 0 ; i < key.length ; i++ ) {
+
+            // step 1 - extract column & sum
+            double  sum = 0;
+            for ( int j = 0 ; j < probCol.length ; j++ ) {
+                sum += (probCol[j] = flowRead.getProb(i, j));
+            }
+
+            // step 2 - normalize column
+            for ( int j = 0 ; j < probCol.length ; j++ ) {
+                probCol[j] /= sum;
```

Comment:
is `sum` ever still 0 here?

Reply:
Addressed (although I don't think real data will ever produce 0)